spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/ColumnRef.scala

```diff
@@ -95,3 +95,28 @@ case object RowCountRef extends ColumnRef {
   override def cqlValueName: String = "count"
   override def cql: String = "count(*)"
 }
+
+/** References a function call **/
+case class FunctionCallRef(
+  columnName: String,
+  actualParams: Seq[Either[ColumnRef, String]] = Seq.empty,
+  alias: Option[String] = None) extends ColumnRef {
+
+  override def selectedAs: String = alias.getOrElse(cqlValueName)
+  override def cqlValueName: String = s"$columnName(${stringifyArguments(actualParams, _.cqlValueName)})"
+  override def cql: String = s"$columnName(${stringifyArguments(actualParams, _.cql)})"
+
+  def requiredColumns: Seq[ColumnRef] = actualParams flatMap {
```

Comment:
@RussellSpitzer This method provides a way for the selector to obtain the function call required columns